pynautobot/core/endpoint.py

```diff
@@ -394,6 +394,56 @@ def count(self, *args, api_version=None, **kwargs):
 
         return ret.get_count()
 
+    def run(self, *args, api_version=None, **kwargs):
+        r"""Runs a job based on the class_path provided to the job.
+
+        Takes a kwarg of `class_path` to specify the job that should be run.
+
+        :arg str,optional \*args: Freeform search string that's
+            accepted on given endpoint.
+        :arg str,optional \**kwargs: Any search argument the
+            endpoint accepts can be added as a keyword arg.
+        :arg str,optional api_version: Override default or globally-set
+            Nautobot REST API version for this single request.
+
+        :Returns: Job details: job_result object uuid found at `obj.result.id`.
+
+        :Examples:
+
+        To return a count of objects matching a named argument filter.
+
+        >>> nb.extras.jobs.run(
+                class_path="local/data_quality/VerifyHostnames",
+                data={"hostname_regex": ".*"},
+                commit=True,
+            )
+
+        >>>
+        """
+
+        if "jobs" not in self.url:
+            raise ValueError('The "run" method can only be executed against the "jobs" app.')
+
+        api_version = api_version or self.api.api_version or self.api.version
```

Comment:
Would this make sense in case none of those were found?
```suggestion
        api_version = api_version or self.api.api_version or self.api.version or None
        
        if api_version is None:
            raise Error
```

Reply:
self.api.version is a class property that is returned from querying nautobot itself for the version.  Don't see a way it can be None.